File: api_service/api/post.py
from fastapi import APIRouter

from api.exceptions import HttpNotFoundException
from models.post import FullPost, PostUpdate
from services.post import post_service

router = APIRouter()


@router.get('/{id}/details')
async def get_post_details(
        id: int,
        related: str = None,
) -> FullPost:
    post = await post_service.get_by_id(id)

    if not post:
        raise HttpNotFoundException()

    result = FullPost(
        post=post.post
    )
    if related:
        if 'user' in related:
            result.author = post.author
        if 'forum' in related:
            result.forum = post.forum
        if 'thread' in related:
            result.thread = post.thread
    return result


@router.post('/{id}/details')
async def edit_post(
        id: int,
        item: PostUpdate
):
    post = await post_service.update_by_id(id, item)

    if not post:
        raise HttpNotFoundException()

    return post

File: api_service/api/post.go
package api

import (
	"db_tp/models"
	"db_tp/services"
	"fmt"
	"github.com/mailru/easyjson"
	"github.com/valyala/fasthttp"
	"strconv"
	"strings"
)

func GetPostDetails(ctx *fasthttp.RequestCtx) {
	id, _ := strconv.Atoi(ctx.UserValue("id").(string))
	related := string(ctx.QueryArgs().Peek("related"))

	post := services.PostSrv.GetById(id)

	if post == nil {
		errMsg := models.ErrMsg{Message: fmt.Sprintf("Can't find post with id:  %s", id)}
		response, _ := easyjson.Marshal(errMsg)
		ctx.SetBody(response)
		ctx.SetStatusCode(404)
		ctx.SetContentType("application/json")
		return
	}

	var response models.FullPost
	response.Post = post.Post
	if strings.Contains(related, "user") {
		response.Author = post.Author
	}
	if strings.Contains(related, "forum") {
		response.Forum = post.Forum
	}
	if strings.Contains(related, "thread") {
		response.Thread = post.Thread
	}

	resp, _ := easyjson.Marshal(response)
	ctx.Response.SetBody(resp)
	ctx.SetContentType("application/json")
	ctx.Response.SetStatusCode(200)
}

//func EditPost(ctx *fasthttp.RequestCtx) {
//
//}
